SocialService/Social/LinkedinService.py

```diff
@@ -108,11 +105,102 @@ async def user_list(self, ai: str):
             print(f"{user['username']}:")
             print(Ai.generate_prompt_linkedin(profile, posts, generation_option, payload, ai, template, api_key=""))
 
+    async def single_company_cli(self, company: str, output: str, payload: str, ai: str, template: str, api_key: str = ""):
+        profile, posts = self._get_company(company)
+        if profile == None:
+            print(f"{Const.COLOR_ERROR}User not found{Const.RESET_ALL}")
+            return
+        
+        Ai = AIRender(ai).model
+
+        print(Ai.generate_prompt_linkedin(profile, posts, output, payload, ai, template, api_key=api_key))
+        sys.exit()
+
+    async def company_list_cli(self, file_path: str, output: str, payload: str, ai: str, template: str, api_key: str = ""):
+        try:
+            with open(file_path, 'r') as file:
+                companies = file.readlines()
+        except:
+            print(f"{Const.COLOR_ERROR}File not found{Const.RESET_ALL}")
+            return
+        clean_companies= []
+        for company in companies:
+            if '\n' in company:
+                company = company[:-1]
+            clean_companies.append(company)
+        profile_list = []
+        updates_list = []
+        for comp_id in clean_companies:
+            profile, updates = self._get_company(comp_id)
+            if profile == None:
+                print(f"{Const.COLOR_ERROR}User {comp_id} not found{Const.RESET_ALL}")
+                profile_list.append(None)
+                updates_list.append(None)
+                continue
+            profile_list.append(profile)
+            updates_list.append(updates)
+
+        Ai = AIRender(ai).model
+
+        for i, (profile, update) in enumerate(zip(profile_list, updates_list)):
+            print(f"{companies[i].title()}:")
+            print(Ai.generate_prompt_linkedin(profile, update, output, payload, ai, template, api_key=api_key))
+
+    async def single_company(self, ai: str):
+        print("\n\nPlease enter the user ID of the user you would like to scrape:")
+        print("(the user ID is in the URL of the user's profile like this: https://www.linkedin.com/company/user-id/)")
+        user_id = input("\Company ID: ")
+        company, updates = self._get_company(user_id)
+        if company == None:
+            print(f"{Const.COLOR_ERROR}Company not found{Const.RESET_ALL}")
+            return self.single_company()
+        generation_option = self._linkedin_generation_options()
+        payload, payload_text = self._payload_options()
+        template = self._template_options()
+
+        Ai = AIRender(ai).model
+
+        print(Ai.generate_prompt_linkedin(company, updates, generation_option, payload, ai, template, payload_text, api_key=""))
+
+    async def company_list(self, ai: str):
+        print("\n\nPlease enter the path to the file containing the list of usernames you would like to scrape:")
```

Comment:
here as well `Please enter the path to the file containing the list of companies you would like to scrape:`